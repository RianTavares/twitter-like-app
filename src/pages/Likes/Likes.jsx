import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import TweetsList from '../../components/TweetsList';
import * as PageActions from '../../store/actions/pages';
import { PAGE_NAMES } from '../../utils/constants';

const Likes = () => {
  const dispatch = useDispatch();
  const likedPost = useSelector((state) => state.tweets.likedTweets);

  useEffect(() => {
    dispatch(PageActions.setCurrentPage({ index: 1, name: PAGE_NAMES.LIKE }));
  }, []);

  if (likedPost.length > 0) {
    return (
      <TweetsList type={PAGE_NAMES.LIKE} />
    );
  }

  return <p className="no-liked-tweets">There are no liked tweets to display.</p>;
};

export default Likes;
