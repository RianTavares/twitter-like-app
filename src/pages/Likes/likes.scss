.no-liked-tweets {
  text-align: center;
  font-style: italic;
  margin-top: $GUTTER;
}