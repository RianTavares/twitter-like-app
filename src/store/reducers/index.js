import { combineReducers } from 'redux';
import pages from './pages';
import tweets from './tweets';

export default combineReducers({
  tweets,
  pages,
});
