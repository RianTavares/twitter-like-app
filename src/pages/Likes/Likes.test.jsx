import '@testing-library/jest-dom';
import { render as rtlRender, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import store from '../../store';
import Likes from './index';

const mockStore = configureStore([]);
const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Likes page', () => {
  test('renders message before any tweet is liked', async () => {
    render(<Likes />);

    expect(screen.getByText('There are no liked tweets to display.')).toBeInTheDocument();
  });
});

describe('Likes page', () => {
  let _store;
  let component;
  const observers = [{
    id: `iamdevloper-${Date.now()}`,
    liked: false,
    account: 'iamdevloper',
    timestamp: Date.now(),
    content: 'Expert Tweet number 1',
  }];
  beforeEach(() => {
    _store = mockStore({
      tweets: {
        likedTweets: observers,
      },
    });

    component = renderer.create(
      <Provider store={_store}>
        <Likes />
      </Provider>,
    );
  });

  test('renders tweets liked list structure', async () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
