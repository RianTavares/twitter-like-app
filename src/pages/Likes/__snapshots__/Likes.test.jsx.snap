// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Likes page renders tweets liked list structure 1`] = `
<ul
  className="tweets-list"
>
  <li>
    <article
      className="tweet"
    >
      <section
        className="profile-picture"
      >
        <div
          className="profile-picture__shape"
        />
      </section>
      <section
        className="tweet__content"
      >
        <div
          className="tweet__header"
        >
          <span
            className="tweet__account"
          >
            iamdevloper
          </span>
          <span
            className="tweet-published-time"
          >
            a second ago
          </span>
        </div>
        <p
          className="tweet__text"
        >
          Expert Tweet number 1
        </p>
        <div
          className="tweet__actions"
        >
          <button
            className="like-canvas"
            onClick={[Function]}
            type="button"
          >
            <div
              className="like-canvas__heart"
            />
          </button>
        </div>
      </section>
    </article>
  </li>
</ul>
`;
